### PR TITLE
Contrast security hook

### PR DESCRIPTION
**Pull request by gmoore**
* A short explanation of the proposed change:

Contrast Security provides agents that our customers can inject into their code in order to both discover vulnerabilities and block attacks. This change makes it easier for NodeJS applications to integrate with Contrast Security.

* An explanation of the use cases your change solves

Contrast Security already integrates with Cloud Foundry through our code in the Java Buildpack as well as our tile available to PCF and Bluemix users. The use case that this change solves is that an operations engineer can configure Contrast Security credentials for an entire org or space, and applications pushed to that space can consume those credentials without any extra configuration on their part. Without this integration, application developers must manage different credentials for different environments themselves. 

* [x] I have viewed signed and have submitted the Contributor License Agreement

* [x] I have made this pull request to the `develop` branch

* [x] I have added an integration test

### CODE REVIEW

**Comment by gmoore** [comment]
@prodion23 @astone123 FYI

File: src/nodejs/integration/nodejs_app_with_contrast_security_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    app = DestroyApp(app)
+
+    RunCF("purge-service-offering", "-f", serviceOffering)
+    RunCF("delete-service-broker", "-f", serviceOffering)
+    RunCF("delete-service", "-f", serviceName)
+
+    serviceBrokerApp = DestroyApp(serviceBrokerApp)
+  })
+
+  FIt("deploying a NodeJS app with Contrast Security", func() {
```
Comment (tylerphelan) [inline src/nodejs/integration/nodejs_app_with_contrast_security_test.go]: @gmoore  Could you take out the `FIt`?

Comment (sclevine) [inline src/nodejs/integration/nodejs_app_with_contrast_security_test.go]: Remove focus.

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,4 @@
 /nodejs_buildpack-*.zip
 /.bin
 /pkg/
+buildpack.sh
```
Comment (sclevine) [inline .gitignore]: Seems unnecessary to add this to gitignore.

File: src/nodejs/hooks/contrast_security.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+  if(!success) {
+    h.Log.Info("Contrast Security no credentials found. Will not write environment files.")
+    return nil
+  }
+
+  profileDDir := path.Join(stager.BuildDir(), ".profile.d")
+
+  if _, err := os.Stat(profileDDir); os.IsNotExist(err) {
+    os.Mkdir(profileDDir, 0755)
```
Comment (sclevine) [inline src/nodejs/hooks/contrast_security.go]: Use `0777`/`0666` and trust OS umask to adjust.

**Comment by tylerphelan** [comment]
@gmoore thanks for the PR! Looks great. After fixing the suggested changes we will prioritize this to merge it in.

**Comment by gmoore** [comment]
Hey @tylerphelan I am up to date on comments. Now I'm seeing merge conflicts though in the PR. What changed?

**Pull request closed without merging**